src/Tools/ExternalAccess/LiveShare/Classification/RoslynSyntaxClassificationService.cs

```diff
@@ -0,0 +1,192 @@
+            });
+        }
+
+        public async Task AddSemanticClassificationsAsync(Document document, TextSpan textSpan, Func<SyntaxNode, ImmutableArray<ISyntaxClassifier>> getNodeClassifiers, Func<SyntaxToken, ImmutableArray<ISyntaxClassifier>> getTokenClassifiers, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+            // If we are in the preview workspace, the request is to try to colorize a lightbulb preview. The document would have changed
+            // in this workspace and we currently don't support requests for anything but the Workspace.CurrentSolution.
+            if (document.Project.Solution.Workspace.Kind == WorkspaceKind.Preview)
+            {
+                return;
+            }
+
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            await AddRemoteClassificationsAsync(RoslynMethods.ClassificationsName, document.FilePath, sourceText, textSpan, result, cancellationToken).ConfigureAwait(false);
+        }
+
+        public void AddSyntacticClassifications(SyntaxTree syntaxTree, TextSpan textSpan, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+            if (_experimentationService.IsExperimentEnabled(WellKnownExperimentNames.SyntacticExp_Local))
+            {
+                using (new RequestLatencyTracker(SyntacticLspLogger.RequestType.SyntacticClassifications))
+                {
+                    _originalService.AddSyntacticClassifications(syntaxTree, textSpan, result, cancellationToken);
+                }
+            }
+            else if (_experimentationService.IsExperimentEnabled(WellKnownExperimentNames.SyntacticExp_Remote))
+            {
+                using (new RequestLatencyTracker(SyntacticLspLogger.RequestType.SyntacticClassifications))
+                {
+                    _threadingContext.JoinableTaskFactory.Run(async () =>
+                    {
+                        var sourceText = await syntaxTree.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        await AddRemoteClassificationsAsync(SyntaxClassificationsHandler.SyntaxClassificationsMethodName, syntaxTree.FilePath, sourceText, textSpan, result, cancellationToken).ConfigureAwait(false);
+                    });
+                }
+            }
+            else
+            {
+                // Something wrong with experiment data.  Just fallback to the regular service.  Don't want to block the user based on an experimentation failure.
+                _originalService.AddSyntacticClassifications(syntaxTree, textSpan, result, cancellationToken);
+            }
+        }
+
+        public ClassifiedSpan FixClassification(SourceText text, ClassifiedSpan classifiedSpan)
+        {
+            return _originalService.FixClassification(text, classifiedSpan);
```

Comment:
ah, i see.  you still keep the lexical fixup local to deal with problems with the results coming back too slowly.